File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day06Test.java
package com.capital7software.aoc.aoc2023.days;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day06Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = LoggerFactory.getLogger(Day06Test.class);

    @BeforeEach
    void setUp() {
        var instance = new Day06();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void productOfTheNumberOfWinningHoldTimesPerRace() {
        var instance = new Day06();
        int expected = 288;
        var actual = instance.productOfTheNumberOfWinningHoldTimesPerRace(lines);
        assertEquals(expected, actual, "The product of the number of winning hold times per race " +
                "is not what was expected: " + expected);
    }

    @Test
    void numberOfWinningHoldTimesQuadratic() {
        var instance = new Day06();
        int expected = 71503;
        var actual = instance.numberOfWinningHoldTimesQuadratic(lines);
        assertEquals(expected, actual, "The number of winning hold times quadratically " +
                "is not what was expected: " + expected);
    }

    @Test
    void numberOfWinningHoldTimesIterative() {
        var instance = new Day06();
        int expected = 71503;
        var actual = instance.numberOfWinningHoldTimesIterative(lines);
        assertEquals(expected, actual, "The number of winning hold times iteratively " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2024/src/test/kotlin/com/capital7software/aoc/aoc2024aoc/days/Day06Test.kt
package com.capital7software.aoc.aoc2024aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day06Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day06Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day06()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testDistinctNumberOfGuardPositions() {
    val instance = Day06()
    val expected = 41

    val actual = instance.distinctNumberOfGuardPositions(lines)
    Assertions.assertEquals(
        expected,
        actual,
        "The actual number of $actual is not the expected number of $expected."
    )
  }

  @Test
  fun testNumberOfPossibleObstructionPositions() {
    val instance = Day06()
    val expected = 6

    val actual = instance.numberOfPossibleObstructionPositions(lines)
    Assertions.assertEquals(
        expected,
        actual,
        "The actual number of $actual is not the expected number of $expected."
    )
  }
}
